cpp/src/parquet/encryption/internal_file_decryptor.cc

```diff
@@ -64,17 +66,20 @@ InternalFileDecryptor::InternalFileDecryptor(FileDecryptionProperties* propertie
   properties_->set_utilized();
 }
 
+InternalFileDecryptor::~InternalFileDecryptor() { WipeOutDecryptionKeys(); }
+
 void InternalFileDecryptor::WipeOutDecryptionKeys() {
-  std::lock_guard<std::mutex> lock(mutex_);
+  std::unique_lock lock(mutex_);
   properties_->WipeOutDecryptionKeys();
-  for (auto const& i : all_decryptors_) {
-    if (auto aes_decryptor = i.lock()) {
-      aes_decryptor->WipeOut();
-    }
-  }
+  footer_key_.clear();
```

Comment:
@pitrou why do we clear the footer key? Do we not trust the string destructor to wipe memory before releasing it?

The same reason probably why we wipe out keys in `properties`.

Reply:
It looks like `std::string::clear()` sets the length of the string to 0, so the string value becomes inaccessible, but the key is not erased in the memory. No need to call `footer_key.clear()` from destructor because the string will be deleted anyways.